trace-forward/src/Trace/Forward/Utils.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Trace.Forward.Utils
+  ( runActionInLoop
+  ) where
+
+import           Control.Exception (SomeAsyncException (..), fromException, tryJust)
+import           Control.Tracer (showTracing, stdoutTracer, traceWith)
+import           System.Time.Extra (sleep)
+
+import           Trace.Forward.Configuration (HowToConnect)
+
+-- | Run monadic action in a loop. If there's an exception, it will re-run
+--   the action again, after pause that grows.
+runActionInLoop
+  :: IO ()
+  -> HowToConnect
+  -> Word
+  -> IO ()
+runActionInLoop action endpoint prevDelay =
+  tryJust excludeAsyncExceptions action >>= \case
+    Left e -> do
+      logTrace $ "trace-forward, connection with " <> show endpoint <> " failed: " <> show e
+      let currentDelay = mkDelayInSecs
```

Comment:
This `let` is unnecesary :); I'd just rename `mkDelayInSecs` to `currentDelay` and use that instead.

Reply:
Thanks, removed.